process/transactions/transactionsGrouper.go

```diff
@@ -244,33 +244,45 @@ func convertMapTxsToSlice(txs map[string]*data.Transaction) []*data.Transaction
 	return transactions
 }
 
-func addToAlteredAddresses(
+func (tg *txsGrouper) addToAlteredAddresses(
 	tx *data.Transaction,
 	alteredAddresses map[string]*data.AlteredAccount,
 	miniBlock *block.MiniBlock,
 	selfShardID uint32,
 	isRewardTx bool,
 ) {
-	isESDTTx := tx.EsdtTokenIdentifier != "" && tx.EsdtValue != ""
+	var nftNonceSTR string
```

Comment:
duplicated code with transactionDBBuilder.go L188

Reply:
refactored. extract this logic in a new function.